src/lib/utils/thorchain/memo/addLimitToMemo.ts

```diff
@@ -0,0 +1,64 @@
+import { assertAndProcessMemo } from './assertAndProcessMemo'
+import { MEMO_PART_DELIMITER } from './constants'
+
+type AddLimitToMemoArgs = {
+  // A THORChain memo, with or without a limit/MinOut
+  memo: string
+  // The limit/minout to be added to be added to the memo. Must be in THOR base unit, or problems.
+  limit: string
+}
+
+// A simple util that adds a limit to a THORChain memo, if the memo doesn't already have one.
+// This is made to be composable, do one thing, and do it well.
+// If you need to handle slippage concerns, this should be done beforehand, with addLimitToMemo/assertAndProcessMemo being the final steps.
+export const addLimitToMemo = ({ memo, limit }: AddLimitToMemoArgs) => {
+  const memoParts = memo.split(MEMO_PART_DELIMITER)
+  const [action] = memoParts
+
+  if (!action) throw new Error(`action is required in memo: ${memo}`)
+
+  let updatedMemo = memo
+
+  switch (action.toLowerCase()) {
+    case 'swap':
+    case '=':
+    case 's': {
+      // SWAP:ASSET:DESTADDR:LIM/INTERVAL/QUANTITY:AFFILIATE:FEE
+      const [, asset, destAddr, _limit, affiliate, fee] = memoParts
```

Comment:
can declare these once [when action is declared](https://github.com/shapeshift/web/pull/6890/files#diff-493036ab0609840c3ae8f999ddb29561ad723e5c3f5110469c914c3569dd35b5R16) to avoid need for closure hoisting workaround

Reply:
then the other 2 redeclarations can be removed below